examples/chip-tool/commands/common/Command.cpp

```diff
@@ -91,10 +95,65 @@ bool Command::InitArgument(size_t argIndex, const char * argValue)
         break;
     }
 
-    case ArgumentType::String: {
+    case ArgumentType::CharString: {
         const char ** value = reinterpret_cast<const char **>(arg.value);
-        *value              = const_cast<char *>(argValue);
-        isValidArgument     = (strcmp(argValue, *value) == 0);
+        *value              = argValue;
+        isValidArgument     = true;
+        break;
+    }
+
+    case ArgumentType::OctetString: {
+        auto * value = static_cast<chip::ByteSpan *>(arg.value);
+        // We support two ways to pass an octet string argument.  If it happens
```

Comment:
Since it's easier to have discussions inline in the code, I've moved my comment here.

The canonical unix solution to this is to receive the data from stdin.  I'm not saying this to troll anybody.  This is a serious suggestion.

As this PR makes clear, it's difficult to pass arbitrary data as a command line argument.  But it's trivial to do so from stdin.  So why don't we just make stdin the pathway for passing in binary things like the thread operational dataset?

Reply:
We can do that.  But some commands (e.g. ble-wifi pairing) take multiple binary things as args.  So we'd need some sort of separators in our stdin stream... and we would need something to generate that stream in the end anyway.

We can do that if it's a good idea, but it's a pretty big rearchitecting of how chip-tool works.  Would we move all arguments to stdin, or just the octet strings?  It would probably need to be all unless we have a command-line arg placeholder for "read from stdin here"....

Reply:
Fair enough.